Core/include/Acts/Surfaces/detail/IntersectionHelper2D.hpp

```diff
@@ -47,11 +49,27 @@ struct IntersectionHelper2D {
   /// @param direction The Direction of intersection line
   ///
   /// @return the intersection points
-  static inline std::pair<Intersection2D, Intersection2D> intersectCircle(
+  static inline std::array<Intersection2D, 2> intersectCircle(
       double R, const Vector2D& origin, const Vector2D& dir) {
     return intersectEllipse(R, R, origin, dir);
   }
 
+  /// Intersect a circle segment
+  ///
+  /// @note only forward solution is taken
+  ///
+  /// @param R The radius
+  /// @param phiMin The minimum phi value
+  /// @param phiMax The maximum phi value
+  /// @param origin The Start of intersection line
+  /// @param direction The Direction of intersection line
+  ///
+  /// @return the intersection points
+  static Intersection2D intersectCircleSegment(double R, double phiMin,
```

Comment:
Don't we need the boundary check for the intersecting?

Reply:
By `segment` the boundary check is implicit.